src/main/java/io/neonbee/endpoint/health/HealthCheckHandler.java

```diff
@@ -1,64 +1,168 @@
+ * health check name is passed in the request path, only health checks for the single health check are returned. By
+ * default, the HTTP response is verbose and provide detailed information about every single health check. If the
+ * verbose flag is set to false via the constructor, only the overall status and a version number is returned.
+ */
 public class HealthCheckHandler implements Handler<RoutingContext> {
     @VisibleForTesting
     static final String APPLICATION_JSON_CHARSET_UTF_8 = "application/json;charset=UTF-8";
 
+    @VisibleForTesting
+    static final String VERSION_FILE = "META-INF/neonbee/neonbee-version.txt";
+
+    private static final String DEV_VERSION = "0.0.0+dev";
+
     private static final LoggingFacade LOGGER = LoggingFacade.create();
 
     private final HealthCheckRegistry registry;
 
+    private final boolean verbose;
+
+    private final Vertx vertx;
+
+    private final AtomicReference<String> version = new AtomicReference<>();
+
+    /**
+     * Constructs an instance of {@link HealthCheckHandler}.
+     *
+     * @param registry a health check registry
+     * @param vertx    the current Vert.x instance
+     */
+    public HealthCheckHandler(HealthCheckRegistry registry, Vertx vertx) {
+        this(registry, true, vertx);
+    }
+
     /**
      * Constructs an instance of {@link HealthCheckHandler}.
      *
      * @param registry a health check registry
+     * @param verbose  whether response should contain verbose health information or not
+     * @param vertx    the current Vert.x instance
+     * @see HealthCheckHandler#HealthCheckHandler(HealthCheckRegistry, Vertx)
      */
-    public HealthCheckHandler(HealthCheckRegistry registry) {
+    HealthCheckHandler(HealthCheckRegistry registry, boolean verbose, Vertx vertx) {
         this.registry = registry;
+        this.verbose = verbose;
+        this.vertx = vertx;
     }
 
     @Override
     public void handle(RoutingContext rc) {
-        registry.collectHealthCheckResults(new DataContextImpl(rc)).onSuccess(resp -> handleRequest(rc, resp))
-                .onFailure(t -> handleFailure(rc, t));
+        String checkName = parseCheck(rc);
+        registry.collectHealthCheckResults(new DataContextImpl(rc), checkName)
+                .onFailure(t -> handleFailure(rc, t))
+                .compose(resp -> handleRequest(rc, resp, verbose));
     }
 
-    private static void handleRequest(RoutingContext rc, JsonObject json) {
+    /**
+     * Parses the health check name from the request path if provided.
+     *
+     * @param rc the {@link RoutingContext} with the request
+     * @return the health check name if provided, empty String otherwise
+     */
+    private String parseCheck(RoutingContext rc) {
+        String requestPath = rc.request().path();
+        String routePath =
+                Optional.ofNullable(rc.currentRoute()).map(Route::getPath).orElse(EMPTY).replaceAll("/+$", EMPTY) + "/";
```

Comment:
```suggestion
        String routePath = Optional.ofNullable(rc.currentRoute())
                .map(Route::getPath)
                .map(path -> path.replaceAll("/+$", EMPTY) + "/")
                .orElse("/");
```
This way we execute replaceAll only on a non null path.

Reply:
that's better!